### ISSUE TITLE
reading in data from ALFAM2 database

### ISSUE THREAD

**Issue by JohannaPedersen** [AU-BCE-EE/Pedersen-2023-sep-and-plasma-treat#1]
https://github.com/AU-BCE-EE/Pedersen-2023-sep-and-plasma-treat/blob/6270136bee5fbd76a07c61b834f5511e5e83fb0b/data-emission/scripts/load.R#L3

You helped me with this before, but now I cannot make it work, can you see what I'm doing wrong @sashahafner ?

**Comment by sashahafner** [comment]
Sure. 
Problem is I have note created a v2.44 release yet. 
I'm waiting for some details from NL (especially incorporation timing).

Solution is to use commit hash instead.

So this should work:
```
ghpath <- 'https://github.com/sashahafner/ALFAM2-data/raw/adb3b8c6a745dc91062cf0f0020d63492b604b27/'
rtag <- ''
...
```

`rtag` is empty but good to keep for switching over once there is a new release.

**Comment by JohannaPedersen** [comment]
Thank you. Silly mistake from me. Now it is working. 

Can you send a screen dump or description of how exactly I find the link to use in ghpath? When I insert the link above into my browser I get 404 error.

**Comment by sashahafner** [comment]
It isn't very clear, and I am sure the approach could be improved.
Here is how to find that special URL for any file on GitHub.

1. Take URL you get when browsing to file through GitHub, e.g., <https://github.com/AU-BCE-EE/STM/blob/master/README.md>.

2a. For commit-based link, replace blob/master with raw/commit_hash, e.g., raw/32f00c46cb75842c8f9dc85d38f23705e3bcd027 for <https://github.com/AU-BCE-EE/STM/raw/32f00c46cb75842c8f9dc85d38f23705e3bcd027/README.md>. 

2b. For release-based link, replace blob/master with raw/release_tag, e.g.,  <https://github.com/AU-BCE-EE/STM/raw/v1.0/README.md>.

You should be able to use either URL in a browser to see the file contents, or download the file when you enter it.
The "raw" bit gives file contents without any formatting.

I don't know where I originally found this info, but if you copy the link that you get with the "raw" button on any file you get a URL with the "raw" bit. 

The reason you can't enter this URL below in a browser is just because it is incomplete--it doesn't have the rest of the file info.
I set up the scripts like that so this only needs to be changed in one place and then I can use this part of the URL with `paste0()` to download two different files.

```
ghpath <- 'https://github.com/sashahafner/ALFAM2-data/raw/adb3b8c6a745dc91062cf0f0020d63492b604b27/'
```

**Comment by sashahafner** [comment]
By the way, I released v2.45 of ALFAM2-data last week so this should now work:

```
# Set release tag for download
ghpath <- 'https://github.com/sashahafner/ALFAM2-data/raw/'
rtag <- 'v2.45'
```

Also I will move this repo to AU-BCE-EE soon but the old URLs should still work.

**Comment by JohannaPedersen** [comment]
Thanks! I'm not sure it is intuitive to me, but now I know how to do it :)
